src/Services/ConfigurationHandler.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\PDGBundle\Services;
+
+use ApiPlatform\PDGBundle\DependencyInjection\Configuration;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Read user configuration on runtime.
+ */
+final class ConfigurationHandler
+{
+    private ?array $config = null;
+
+    public function get(string $name, $default = null): mixed
+    {
+        $this->parse();
+
+        // Convert "foo.bar" in "['foo' => ['bar' => ...]]"
+        $config = $this->config;
+        $keys = explode('.', $name);
+        foreach ($keys as $key) {
+            if (\array_key_exists($key, $config)) {
+                $config = $config[$key];
+                continue;
+            }
+
+            return $default;
+        }
+
+        return $config;
+    }
+
+    private function parse(): void
+    {
+        // First, load config file from PHP_DOC_CONFIG environment variable
+        $configFile = getenv('PHP_DOC_CONFIG');
```

Comment:
Maybe rename 'PHP_DOC_CONFIG' to 'PDG_CONFIG' for consistency with the file name

Reply:
`PDG_CONFIG_FILE` would be even more consistent